### ISSUE TITLE
Strange bug upon preemption

### ISSUE THREAD

**Issue by sdascoli** [facebookincubator/submitit#1613]
Hi, since the cluster update I am encoutering the following error upon preemption:
```RuntimeError: Timed out initializing process group in store based barrier on rank: 0, for key: store_based_barrier_key:1 (world_size=8, worker_count=16, timeout=0:30:00)```

Full log:
```slurmstepd: error: *** STEP 37361369.0 ON learnfair5146 CANCELLED AT 2021-03-15T17:50:23 ***
submitit WARNING (2021-03-15 17:50:23,701) - Bypassing signal 15
WARNING:submitit:Bypassing signal 15
slurmstepd: error: *** JOB 37361369 ON learnfair5146 CANCELLED AT 2021-03-15T18:00:46 DUE TO PREEMPTION ***
srun: Job step aborted: Waiting up to 32 seconds for job step to finish.
submitit WARNING (2021-03-15 18:00:46,902) - Bypassing signal 15
WARNING:submitit:Bypassing signal 15
slurmstepd: error: Failed to send MESSAGE_TASK_EXIT: Connection refused
submitit ERROR (2021-03-15 18:45:40,210) - Submitted job triggered an exception
ERROR:submitit:Submitted job triggered an exception
Traceback (most recent call last):
  File "/private/home/sdascoli/.conda/envs/bert/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/private/home/sdascoli/.conda/envs/bert/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/private/home/sdascoli/.conda/envs/bert/lib/python3.7/site-packages/submitit/core/_submit.py", line 11, in <module>
    submitit_main()
  File "/private/home/sdascoli/.conda/envs/bert/lib/python3.7/site-packages/submitit/core/submission.py", line 71, in submitit_main
    process_job(args.folder)
  File "/private/home/sdascoli/.conda/envs/bert/lib/python3.7/site-packages/submitit/core/submission.py", line 64, in process_job
    raise error
  File "/private/home/sdascoli/.conda/envs/bert/lib/python3.7/site-packages/submitit/core/submission.py", line 53, in process_job
    result = delayed.result()
  File "/private/home/sdascoli/.conda/envs/bert/lib/python3.7/site-packages/submitit/core/utils.py", line 126, in result
    self._result = self.function(*self.args, **self.kwargs)
  File "run_pretrained.py", line 64, in __call__
    classification.main(self.args)
  File "/checkpoint/sdascoli/cnn2transformer/1615824683/main.py", line 199, in main
    utils.init_distributed_mode(args)
  File "/checkpoint/sdascoli/cnn2transformer/1615824683/utils.py", line 276, in init_distributed_mode
    world_size=args.world_size, rank=args.rank)
  File "/private/home/sdascoli/.conda/envs/bert/lib/python3.7/site-packages/torch/distributed/distributed_c10d.py", line 525, in init_process_group
    _store_based_barrier(rank, store, timeout)
  File "/private/home/sdascoli/.conda/envs/bert/lib/python3.7/site-packages/torch/distributed/distributed_c10d.py", line 215, in _store_based_barrier
    rank, store_key, world_size, worker_count, timeout))
RuntimeError: Timed out initializing process group in store based barrier on rank: 0, for key: store_based_barrier_key:1 (world_size=8, worker_count=16, timeout=0:30:00)
srun: error: learnfair5094: tasks 0-1,3-7: Exited with exit code 1
srun: launch/slurm: _step_signal: Terminating StepId=37361369.1
slurmstepd: error: *** STEP 37361369.1 ON learnfair5094 CANCELLED AT 2021-03-15T18:45:41 ***
srun: error: learnfair5094: task 2: Exited with exit code 1
```

Any ideas where this can be coming from ?
Thanks :)

**Comment by gwenzek** [comment]
If you're being preempted and still in "init_process_group" then I guess there is a problem with you job config, or the setup code.
Can you share more details ? Job config ? `init_distributed_mode` code ?

**Comment by sdascoli** [comment]
Hi, here is the ``` init_distributed_mode``` code which I imported from the ```deit``` repo:

```def init_distributed_mode(args):
    if 'RANK' in os.environ and 'WORLD_SIZE' in os.environ:
        args.rank = int(os.environ["RANK"])
        args.world_size = int(os.environ['WORLD_SIZE'])
        args.gpu = int(os.environ['LOCAL_RANK'])
    elif 'SLURM_PROCID' in os.environ:
        args.rank = int(os.environ['SLURM_PROCID'])
	args.gpu = args.rank % torch.cuda.device_count()
    else:
        print('Not using distributed mode')
        args.distributed = False
	return

    args.distributed = True
    torch.cuda.set_device(args.gpu)
    args.dist_backend = 'nccl'
    print('| distributed init (rank {}): {}'.format(
        args.rank, args.dist_url), flush=True)
    torch.distributed.init_process_group(backend=args.dist_backend, init_method=args.dist_url,
                                         world_size=args.world_size, rank=args.rank)
    torch.distributed.barrier()
    setup_for_distributed(args.rank == 0)
```

The job config is also inhereted from https://github.com/facebookresearch/deit/blob/main/run_with_submitit.py

Thanks !

**Comment by gwenzek** [comment]
I believe this issue has been solved with #1615 and the 1.3.0 release that was needed after a change in our cluster config. Please reopen if you are still blocked.
